### PR TITLE
Delete Blob Servlet

### LINKED ISSUE #89: Delete unused PDFs from Blobstore
Before deleting a Match entity when a review is finished, make sure we delete the blob in Blobstore associated with the Blob key stored in the Match entity.

### PR DESCRIPTION
**Pull request by shayla11**
The PR is the delete blob servlet. We will take the resumeBlobKey from the response (Match datastore is where the blobkey is currently being stored), and then use blobstoreServive.delete() to delete the blob. I started writing the code to set the status if it successfully deletes the blobstore. I tried to set the status, but could not set a message. Also I am unsure how to check if the blob has actually been deleted. I made an if statement to check if the blobkey is now null, but I don't know whats actually happening to the blobkey once we delete the blob.

### CODE REVIEW

File: resume-buddy/src/main/java/com/google/sps/servlets/DeleteBlobServlet.java [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+
+    String resumeBlobKey = request.getParameter("resumeBlobKey");
+    BlobKey blobKey = new BlobKey(resumeBlobKey);
+    blobstoreService.delete(blobKey);
+
+    // check if blob is empty (not sure if this will work, I dont know what the blobkey is being set
+    // too after its deleted)
+    if (blobKey == null) {
+      isBlobDeleted = true;
+    }
```
Comment (oydang) [inline resume-buddy/src/main/java/com/google/sps/servlets/DeleteBlobServlet.java]: Looks like the way to check if the blob no longer exists is to use [`BlobInfoFactory.loadBlobInfo`](https://cloud.google.com/appengine/docs/standard/java/javadoc/com/google/appengine/api/blobstore/BlobInfoFactory.html#loadblobinfo). If the return value is null, the blob no longer exists.

``` 
BlobInfoFactory blobInfoFactory = new BlobInfoFactory();
isBlobDeleted = blobInfoFactory.loadBlobInfo(blobKey) == null;
```

File: resume-buddy/src/main/java/com/google/sps/servlets/DeleteBlobServlet.java [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+
+    // set the status message corresponding to wheter its true or false
+    if (isBlobDeleted) {
+      response.setStatus(200); // works
+      // response.setMessage("The blob has been deleted!");
```
Comment (oydang) [inline resume-buddy/src/main/java/com/google/sps/servlets/DeleteBlobServlet.java]: Delete this

File: resume-buddy/src/main/java/com/google/sps/servlets/DeleteBlobServlet.java [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+    } else {
+      response.setStatus(400); // error
+      // response.statusText("The blob is not been deleted.");
```
Comment (oydang) [inline resume-buddy/src/main/java/com/google/sps/servlets/DeleteBlobServlet.java]: Delete this

**Review by oydang — CHANGES_REQUESTED** [review]

File: resume-buddy/src/main/java/com/google/sps/servlets/DeleteBlobServlet.java [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    BlobstoreService blobstoreService = BlobstoreServiceFactory.getBlobstoreService();
+    boolean isBlobDeleted = false;
+
+    String resumeBlobKey = request.getParameter("resumeBlobKey");
+    BlobKey blobKey = new BlobKey(resumeBlobKey);
+    blobstoreService.delete(blobKey);
+
+    BlobInfoFactory blobInfoFactory = new BlobInfoFactory();
+    isBlobDeleted = blobInfoFactory.loadBlobInfo(blobKey) == null;
```
Comment (oydang) [inline resume-buddy/src/main/java/com/google/sps/servlets/DeleteBlobServlet.java:24]: Where did the code go that puts a status code 200 or 400 in the response, to let the caller know whether the delete succeeded or not? That was great.

**Review by oydang — CHANGES_REQUESTED** [review]

File: resume-buddy/src/main/java/com/google/sps/servlets/DeleteBlobServlet.java (hunk 4 above)
Comment (shayla11) [reply resume-buddy/src/main/java/com/google/sps/servlets/DeleteBlobServlet.java:24]: When I saw your changes I though it was to delete the whole section, not just the line 💀. It should be back now!

**Review by oydang — APPROVED** [review]
Nice work. The last thing to do is to actually use this servlet to delete blobs when a review is done. Make sure to communicate with Ani, who has been working on the flow that is triggered by the "done reviewing" button being pressed.

**Pull request merged**